requirements.txt

```diff
@@ -49,6 +49,9 @@ git+git://github.com/PyCQA/pylint.git@2561f539d60a3563d6507e7a22e226fb10b58210
 # the tests.  With this here, you can run the tests with `py.test` from the base directory.
 pytest
 
+# Static type checking
+mypy
```

Comment:
Do you think it's worth having a separate `test-requirements.txt` file?  I'm not sure if it is or not, but unless you're running mypy yourself, or running our travis build, our code doesn't strictly require mypy, so it seems a little odd to have it in here.  Also, I think there are some older things above that should be removed, but that's a separate PR.